tests/js/web/PulsarUIComponentTest.js

```diff
@@ -108,6 +134,58 @@ describe('Pulsar UI Component', function() {
         });
     });
 
+    describe('Table row-actions', function() {
+
+        beforeEach(function() {
+            sinon.spy(this.pulsarUIComponent, 'closeRowActions');
+            this.pulsarUIComponent.init();
+        });
+
+        it('should be opened when clicked', function() {
+            this.$rowActions.click();
+            setTimeout(function(){
+                expect(this.$rowActions.parent().hasClass('open')).to.be.true;
+            }, 1000);
+        });
```

Comment:
If we are testing anything async, we need to utilise the `done` callback, otherwise there is a chance your assertions are never ran and reporting a false positive. Anything with a setTimeout needs a `done`:


```
it('should be opened when clicked', function(done) {
    this.$rowActions.click();
    setTimeout(function(){
        expect(this.$rowActions.parent().hasClass('open')).to.be.true;
        done()
    }, 1000);
});
```

Reply:
also on the `setTimeout` used here, how brittle will `1000` be? Does we need to compute this from some CSS or JS animation?

Reply:
I don't event think there's animations on the dropdown, just a delay with how long the event takes, this always trips me up